hoomd/md/methods.py

```diff
@@ -454,69 +454,183 @@ def thermalize_thermostat_and_barostat_dof(self, seed):
+    (with *nph=True*).
+
+    Note:
+        Coupling constant for barostat `tauS` should be set within appropriate
+        range for pressure and volume to fluctuate in reasonable rate and
+        equilibrate. Too small `tauS` can cause abrupt fluctuation, whereas too
+        large `tauS` would take long time to equilibrate. In most of systems,
+        recommended value for `tauS` is ``1000 * dt``, where ``dt`` is the
+        length of the time step.
 
     Examples::
 
-        # Triclinic unit cell
-        nph=integrate.nph(group=all, P=2.0, tauP=1.0, couple="none", all=True)
-        # Cubic unit cell
-        nph = integrate.nph(group=all, P=2.0, tauP=1.0)
-        # Relax the box
-        nph = integrate.nph(group=all, P=0, tauP=1.0, gamma=0.1)
+        nph = hoomd.md.methods.NPH(filter=hoomd.filter.All(), tauS = 1.2, S=2.0)
+        # orthorhombic symmetry
+        nph = hoomd.md.methods.NPH(filter=hoomd.filter.All(), tauS = 1.2, S=2.0, couple="none")
+        # tetragonal symmetry
+        nph = hoomd.md.methods.NPH(filter=hoomd.filter.All(), tauS = 1.2, S=2.0, couple="xy")
+        # triclinic symmetry
+        nph = hoomd.md.methods.NPH(filter=hoomd.filter.All(), tauS = 1.2, S=2.0, couple="none", rescale_all=True)
+        integrator = hoomd.md.Integrator(dt=0.005, methods=[nph], forces=[lj])
+
+    Attributes:
+        filter (hoomd.filter.ParticleFilter): Subset of particles on which to
+            apply this method.
+
+        S (List[hoomd.variant.Variant]): Stress components set
+            point for the barostat (in pressure units).
+            In Voigt notation,
+            :math:`[S_{xx}, S_{yy}, S_{zz}, S_{yz}, S_{xz}, S_{xy}]`. Stress can
+            be reset after method object is created. For example, An isoropic
+            pressure can be set by ``npt.S = 4.``
+
+        tauS (float): Coupling constant for the barostat (in time units).
+
+        couple (str): Couplings of diagonal elements of the stress tensor,
+            can be "none", "xy", "xz","yz", or "all".
+
+        box_dof(List[bool]): Box degrees of freedom with six boolean elements
+            corresponding to x, y, z, xy, xz, yz, each.
+
+        rescale_all (bool): if True, rescale all particles, not only those in
+            the group.
+
+        gamma (float): Dimensionless damping factor for the box degrees of
+            freedom.
+
+        barostat_dof (tuple[float, float, float, float, float, float]):
+            Additional degrees of freedom for the barostat (:math:`\nu_{xx}`,
+            :math:`\nu_{xy}`, :math:`\nu_{xz}`, :math:`\nu_{yy}`,
+            :math:`\nu_{yz}`, :math:`\nu_{zz}`)
     """
-    def __init__(self, **params):
 
-        # initialize base class
-        npt.__init__(self, nph=True, kT=1.0, **params)
+    def __init__(self,
+                 filter,
+                 S,
+                 tauS,
+                 couple,
+                 box_dof=[True, True, True, False, False, False],
```

Comment:
This should default to `None` or a tuple and then we convert or set in the constructor. Default mutable arguments in Python are dangerous as they can mutate in the function itself.